libraries/ms-common/src/stm32f0xx/adc.c

```diff
@@ -0,0 +1,102 @@
+#include "adc.h"
+#include "stm32f0xx.h"  
+
+#include <stdio.h>
+
+static uint8_t prv_get_channel(GPIOAddress *address) {
+  uint8_t adc_channel = address->pin;
+
+  switch (address->port) {
```

Comment:
This seems pretty complex and difficult to maintain. You have a lot of magic numbers floating around.